test_runner/regress/test_branching.py

```diff
@@ -366,9 +365,61 @@ def start_creating_timeline():
 
     wait_until_tenant_active(ps_http, env.initial_tenant)
 
-    # currently it lives on and will get eventually uploaded, but this will change
-    detail = ps_http.timeline_detail(env.initial_tenant, env.initial_timeline)
-    assert detail["state"] == "Active"
+    with pytest.raises(PageserverApiException, match="not found"):
+        ps_http.timeline_detail(env.initial_tenant, env.initial_timeline)
+
+
+def test_non_uploaded_branch_is_deleted_after_restart(neon_env_builder: NeonEnvBuilder):
+    """
+    Currently before RFC#27 we keep and continue uploading branches which were not successfully uploaded before shutdown.
```

Comment:
Is this sentence out-of-date? My understanding of this tests, and the Rust code, is that we delete not-uploaded timeline dirs after restart.

https://github.com/neondatabase/neon/blob/ba118a87824514d92bc407e0d857988cea19138b/pageserver/src/tenant.rs#L1270-L1288